plugins/single_plugins/grid.cpp

```diff
@@ -255,24 +261,69 @@ class wayfire_grid : public wf::plugin_interface_t
             output->add_activator(keys[i], &bindings[i]);
         }
 
+        restore = [=] (wf_activator_source, uint32_t)
+        {
+            auto view = output->get_active_view();
+            restore_view(view);
+        };
+
+        restore_opt = section->get_option("restore", "last");
+        restore_opt_str = restore_opt->as_string();
+        restore_opt_changed = [=] ()
+        {
+            if (restore_opt_str != "last")
+                output->rem_binding(&restore);
+            restore_opt_str = restore_opt->as_string();
+            if (restore_opt_str != "last")
+                output->add_activator(restore_opt, &restore);
+        };
+        restore_opt_changed();
+        restore_opt->add_updated_handler(&restore_opt_changed);
+
         output->connect_signal("reserved-workarea", &on_workarea_changed);
         output->connect_signal("view-snap", &on_snap_signal);
         output->connect_signal("query-snap-geometry", &on_snap_query);
         output->connect_signal("view-maximized-request", &on_maximize_signal);
         output->connect_signal("view-fullscreen-request", &on_fullscreen_signal);
     }
 
+    void restore_view(wayfire_view view)
```

Comment:
Why do we need `restore_view()`? AFAICT `handle_slot(.., slot = 0)` does all we need.